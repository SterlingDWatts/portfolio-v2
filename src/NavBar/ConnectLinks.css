.ConnectLinks {
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
}

@media screen and (min-width: 900px) {
    .ConnectLinks {
        max-width: 944px;
        justify-content: space-between;
        margin: 0 auto;
    }
}
