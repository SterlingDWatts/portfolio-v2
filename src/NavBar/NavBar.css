.NavBar {
    padding: 16px;
    background-color: #003d32;
    color: #fcd734;
    line-height: 1;
    min-height: 48px;
    z-index: 10;
    box-shadow:  6px 6px 10px 5px rgba(0, 0, 0, 0.25);
}

.NavBar a:link, .NavBar a:visited {
    color: #fcd734;
    text-decoration: none;
}

.NavBar:nth-child(1) {
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
}

.NavBar--hidden:nth-child(1) {
    top: -47px;
}