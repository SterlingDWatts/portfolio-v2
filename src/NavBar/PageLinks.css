.PageLinks {
    display: flex;
    flex-direction: row;
    justify-content: space-evenly;
}

.PageLinks li {
    margin-left: 16px;
}

.burger_button {
    border: none;
    padding: 0;
    background-color: rgba(0, 0, 0, 0);
    color: #fcd734;
}