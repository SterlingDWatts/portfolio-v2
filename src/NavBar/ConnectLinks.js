import React, { Component } from 'react';
import './ConnectLinks.css';

class ConnectLinks extends Component {

    render() {
        return (
            <ul className='ConnectLinks'>
                <li><a href="https://github.com/SterlingDWatts" target="_blank"><i className="fab fa-github"></i></a></li>
                <li><a href="https://www.linkedin.com/in/sterlingdwatts/" target="_blank"><i className="fab fa-linkedin-in"></i></a></li>
                <li><a href="https://www.facebook.com/SterlingDWatts" target="_blank"><i className="fab fa-facebook-f"></i></a></li>
                <li><a href="https://www.instagram.com/sterlingdwatts" target="_blank"><i className="fab fa-instagram"></i></a></li>
                <li><a href="https://twitter.com/SterlingDWatts" target="_blank"><i className="fab fa-twitter"></i></a></li>
            </ul>
        );
    }

}

export default ConnectLinks;
