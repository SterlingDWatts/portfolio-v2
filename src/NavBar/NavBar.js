import React, { Component } from 'react';
import './NavBar.css';

class NavBar extends Component {


	render() {
		let classnames = 'NavBar';
		if (this.props.hidden) {
			classnames = 'NavBar NavBar--hidden';
		}
		return (
			<div className={classnames}>
				{this.props.children}
			</div>
		);

	}

}

export default NavBar;
