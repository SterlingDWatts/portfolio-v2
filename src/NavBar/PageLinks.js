import React, { Component } from 'react';
import { Link } from 'react-router-dom';
import './PageLinks.css';

class PageLinks extends Component {

	render() {
		return (
			<ul className='PageLinks'>
				<li className='large'>
					<a href='#about-me'>About Me</a>
				</li>
				<li className='large'>
					<a href='#projects'>Projects</a>
				</li>
				<li className='large'>
					<a href='#connect'>Connect</a>
				</li>
				<li className='large'>
					<Link 
						to='/blog'
					>
						Blog 
					</Link>
				</li>
				<li className='small'>
					<button className='burger_button'

					>
						<i className='fas fa-bars'></i>
					</button>
				</li>
			</ul>
		);
	}

}

export default PageLinks;
