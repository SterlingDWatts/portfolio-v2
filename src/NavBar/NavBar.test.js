import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import NavBar from './NavBar';

describe(`NavBar Component`, () => {

    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<NavBar />, div);
        ReactDOM.unmountComponentAtNode(div);
    })

    it('renders the UI as expected', () => {
        const tree = renderer
            .create(<NavBar />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })

})