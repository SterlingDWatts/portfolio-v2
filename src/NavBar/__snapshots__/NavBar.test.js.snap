// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar Component renders the UI as expected 1`] = `
<div
  className="NavBar"
/>
`;
